cmd/uplink/cmd/setup.go

```diff
@@ -251,3 +260,37 @@ func ApplyDefaultHostAndPortToAddr(address, defaultAddress string) (string, erro
 	// address is host:
 	return net.JoinHostPort(addressParts[0], defaultPort), nil
 }
+
+// SaveEncryptionKey saves the key in a new file which will be stored in
+// filepath.
+// It returns an error if the directory doesn't exist, the file already exists
+// or there is an I/O error.
+func SaveEncryptionKey(key []byte, filepath string) (err error) {
+	f, err := os.OpenFile(filepath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return errors.New("directory path doesn't exist")
```

Comment:
Is this the only case when this can happen? If you know for certain that on all platforms (Windows, Linux, Android, Fuchsia etc.), then this is fine, however when not, then include the original error.

Reply:
Obviously I don't know for 100%

Reply:
Hence include the actual error, I've seen things fail in weird ways. Conceptually, I understand trying to minimize the noise, but we are still in beta and going to hit weird edge cases. So I prefer to have more information rather than cleaner output.

Reply:
I was going to do without further explanation, but I appreciate your message because it gives more context on the current stage.

Reply:
By the way, should we use `errs.Wrap in these errors?